File: abc/abc209/abc209_b.fs
// https://atcoder.jp/contests/abc209/submissions/24164531

let [| N; X |] = stdin.ReadLine().Split() |> Array.map int

stdin.ReadLine().Split() |> Array.map int
|> fun A -> if Array.sum A - N / 2 <= X then "Yes" else "No"
|> stdout.WriteLine

File: abc/abc209/abc209_b.cs
// https://atcoder.jp/contests/abc209/submissions/24162655

using System;
using System.Linq;

class Program
{
    static void Main()
    {
        var input = Console.ReadLine().Split().Select(int.Parse).ToArray();
        var N = input[0];
        var X = input[1];
        var A = Console.ReadLine().Split().Select(int.Parse).ToArray();

        Console.WriteLine(A.Sum() - N / 2 <= X ? "Yes" : "No");
    }
}
